File: Programmers_java/모의고사.java
import java.io.*;
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] answer = new int[3];
        List<Integer> first = new ArrayList<>(Arrays.asList(1, 2, 3, 4, 5));
        List<Integer> second = new ArrayList<>(Arrays.asList(2, 1, 2, 3, 2, 4, 2, 5));
        List<Integer> third = new ArrayList<>(Arrays.asList(3, 3, 1, 1, 2, 2, 4, 4, 5, 5));
        int[] cnt = new int[3];

        while (first.size() < answers.length) {
            first.addAll(first);
        }
        while (second.size() < answers.length) {
            second.addAll(second);
        }
        while (third.size() < answers.length) {
            third.addAll(third);
        }

        for (int i = 0; i < answers.length; i++) {
            if (answers[i] == first.get(i)) cnt[0]++;
            if (answers[i] == second.get(i)) cnt[1]++;
            if (answers[i] == third.get(i)) cnt[2]++;
        }

        int max = 0;
        for (int i = 0; i < 3; i++) {
            max = Math.max(max, cnt[i]);
        }

        int idx = 0;
        for (int i = 0; i < 3; i++) {
            if (max == cnt[i]) {
                answer[idx++] = i + 1;
            }
        }

        answer = Arrays.copyOfRange(answer, 0, idx);
        return answer;
    }
}

File: Programmers_python/모의고사.py
def solution(answers):
    p1 = [1, 2, 3, 4, 5] * 10000
    p2 = [2, 1, 2, 3, 2, 4, 2, 5] * 10000
    p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5] * 10000
    cnt = [0, 0, 0]
    answer = []
    
    for i in range(len(answers)):
        if answers[i] == p1[i]:
            cnt[0] += 1
        if answers[i] == p2[i]:
            cnt[1] += 1
        if answers[i] == p3[i]:
            cnt[2] += 1
            
    for i in range(len(cnt)):
        if cnt[i] == max(cnt):
            answer.append(i+1)
    
    return answer
